DSCResources/MSFT_xWebsite/MSFT_xWebsite.psm1

```diff
@@ -294,6 +337,55 @@ function Set-TargetResource
                     Add-WebConfiguration -filter /system.applicationHost/serviceAutoStartProviders -Value @{name=$ServiceAutoStartProvider; type=$ApplicationType} -ErrorAction Stop
                 }
             }
+            
+            # Update LogFormat if Needed
+            if ($PSBoundParameters.ContainsKey('LogFormat') -and ($LogFormat -ne $Website.logfile.LogFormat))
+            {
+                Write-Verbose -Message ($LocalizedData.VerboseSetTargetUpdateLogFormat -f $Name)
+                Set-ItemProperty -Path "IIS:\Sites\$Name" -Name LogFile.logFormat -value $LogFormat
+            }
+
+            # Update LogFlags if required
+            if ($PSBoundParameters.ContainsKey('LogFlags') -and (-not (Compare-LogFlags -Name $Name -LogFlags $LogFlags))) 
+            {
+                Write-Verbose -Message ($LocalizedData.VerboseSetTargetUpdateLogFlags -f $Name)
+                Set-ItemProperty -Path "IIS:\Sites\$Name" -Name LogFile.logFormat -Value 'W3C'
+                Set-ItemProperty -Path "IIS:\Sites\$Name" -Name LogFile.LogExtFileFlags -Value $LogFlags
+            }
+            
+            #  Update LogPath if required
+            if ($PSBoundParameters.ContainsKey('LogPath') -and ($LogPath -ne $Website.logfile.LogPath))
+            {            
+                Write-Verbose -Message ($LocalizedData.VerboseSetTargetUpdateLogPath -f $Name)
+                Set-ItemProperty -Path "IIS:\Sites\$Name" -Name LogFile.directory -value $LogPath
+            }
+            
+            # Update LogPeriod if needed
+            if ($PSBoundParameters.ContainsKey('LogPeriod') -and ($LogPeriod -ne $Website.logfile.LogPeriod))
+            {
+                if ($PSBoundParameters.ContainsKey('LogTruncateSize'))
+                    {
+                        Write-Verbose -Message ($LocalizedData.WarningLogPeriod -f $Name)
+                    }              
+                Write-Verbose -Message ($LocalizedData.VerboseSetTargetUpdateLogPeriod -f $Name)
+                Set-ItemProperty -Path "IIS:\Sites\$Name" -Name LogFile.period -Value $LogPeriod
+            }
+            
+            # Update LogTruncateSize if needed
+            if ($PSBoundParameters.ContainsKey('LogTruncateSize') -and ($LogTruncateSize -ne $Website.logfile.LogTruncateSize))
```

Comment:
There are a few lines above 100 chars in this code as well as some positional parameters.